pkg/operator/ceph/disruption/clusterdisruption/add.go

```diff
@@ -57,27 +62,78 @@ func Add(mgr manager.Manager, context *controllerconfig.Context) error {
 		return err
 	}
 
+	cephClusterPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			logger.Info("create event from ceph cluster CR")
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldCluster, ok := e.ObjectOld.DeepCopyObject().(*cephv1.CephCluster)
+			if !ok {
+				return false
+			}
+			newCluster, ok := e.ObjectNew.DeepCopyObject().(*cephv1.CephCluster)
+			if !ok {
+				return false
+			}
+			return !reflect.DeepEqual(oldCluster.Spec, newCluster.Spec)
+		},
+	}
+
+	// Watch for CephClusters
+	err = c.Watch(&source.Kind{Type: &cephv1.CephCluster{}}, &handler.EnqueueRequestForObject{}, cephClusterPredicate)
+	if err != nil {
+		return err
+	}
+
+	pdbPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			pdb, ok := e.ObjectNew.DeepCopyObject().(*policyv1beta1.PodDisruptionBudget)
+			if !ok {
+				return false
+			}
+			// only reconcile if allowed disruptions is 0 in the main PDB
+			return pdb.Name == osdPDBAppName && pdb.Status.DisruptionsAllowed == 0
+		},
+	}
+
+	// Watch for main PodDisruptionBudget and enqueue the CephCluster in the namespace
+	err = c.Watch(
+		&source.Kind{Type: &policyv1beta1.PodDisruptionBudget{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+				_, ok := obj.Object.(*policyv1beta1.PodDisruptionBudget)
+				if !ok {
+					// not a pdb, returning empty
+					logger.Errorf("PDB handler received non-PDB")
```

Comment:
Should we return the error instead of just logging it? Or is there a reason we get non-pdb events with this watcher?

Reply:
I don't think there would be case that we would get a no-pdb event for this watcher as its only watching for pdb. We are not reconciling in case there is some error while getting the pdb object.  I wasn't using the pdb object returned in line 105. Update the code to use pdb object returned in line 105